thoughtbot/suspenders
# Support Rails 7 and Ruby 3

# 🚀 Support Rails 7 and Ruby 3

This is a major upgrade that aims to make it so generated Rails applications with suspenders are running on Rails version 7 and Ruby version 3. We took the approach of making the least amount of changes possible.

Special thanks to @emilford, @stefannibrasil and @DoodlingDev for their help!

## 🏗 What changed?

- We removed obsolete and deprecated tooling around CSS #1112 
- We favor [cssbundling-rails](https://github.com/rails/cssbundling-rails) and [jsbundling-rails](https://github.com/rails/jsbundling-rails) over [importmap-rails](https://github.com/rails/importmap-rails) [in order to support more options around CSS frameworks](https://guides.rubyonrails.org/working_with_javascript_in_rails.html#choosing-between-import-maps-and-a-javascript-bundler).

## ❓What next?

1. We'll keep this pull-request open long enough to capture initial feedback. Feel free to add a `to-do` to the list below.
2. Once approved, we'll squash everything down into a single commit, since all these changes are related and can't easily be grouped.
3. Bump the version and update the README. 

## 🙏 How to contribute and review

This helps us determine if suspenders is generating an error free application.

1. Checkout the [support_rails_7](https://github.com/thoughtbot/suspenders/tree/support_rails_7) branch.
5. Once you are in the `suspenders` directory, run `./bin/suspenders some_app_name` to generate a Rails app from the current state of suspenders.
6. Change directories into `some_app_name`.
7. Run `rails g scaffold some_model some_attribute` followed by `rails db:migrate`.
8. Run `rails s`

Open up the app and ensure that it is loaded correctly. Try filing out the forms generated from the scaffold. If you run into an error, please follow the steps below.

1. Open a [new issue](https://github.com/thoughtbot/suspenders/issues/new).
2. Tag it as [Rails 7](https://github.com/thoughtbot/suspenders/labels/Rails%207)
4. Add it to the to-do section below. 

### ✅ To-do

- [x] Investigate `bin/setup` error. (`bin/setup: line 14: bin/yarn: No such file or directory`). Looks like it relates to #939 
- [x] Add normalize via PostCSS
- [x] #1112 which is handled by #1113
- [x] #1116
- [x] #1111 
- [x] #1108
- [x] #1109
